### PR TITLE
Added a `Function6` 

### PR DESCRIPTION
**Pull request by mmscibor**
This has been coming up fairly often recently, as often there will be futures coming in pairs (and I'll have 3 pairs).

### CODE REVIEW

**Comment by dflemstr** [comment]
Lgtm but please don't decrease test coverage!

**Comment by mmscibor** [comment]
@dflemstr added the unit tests to preserve coverage

File: src/main/java/com/spotify/futures/Function6.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+public interface Function6<A, B, C, D, E, F, R> {
+
+  /**
+   * Applies this function to the given arguments.
+   *
+   * @param a the first function argument
+   * @param b the second function argument
+   * @param c the third function argument
+   * @param d the fourth function argument
+   * @param e the firth function argument
```
Comment (mattnworb) [inline src/main/java/com/spotify/futures/Function6.java]: minor: typo in "firth"

File: src/main/java/com/spotify/futures/Function6.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+   * If evaluation of either function throws an exception, it is relayed to
+   * the caller of the composed function.
+   *
+   * @param <V>   the type of output of the {@code after} function, and of the
+   *              composed function
+   * @param after the function to apply after this function is applied
+   * @return a composed function that first applies this function and then
+   * applies the {@code after} function
+   * @throws NullPointerException if after is null
+   * @since 0.1.0
```
Comment (mattnworb) [inline src/main/java/com/spotify/futures/Function6.java]: I think this version number is wrong, maybe it was copy and paste?

File: src/main/java/com/spotify/futures/CompletableFutures.java [hunk 3]
```diff
@@ -383,6 +383,47 @@ public static void checkCompleted(CompletionStage<?> stage) {
...
+   * @param function the combining function.
+   * @param <R>      the type of the combining function's return value.
+   * @param <A>      the type of the first stage's value.
+   * @param <B>      the type of the second stage's value.
+   * @param <C>      the type of the third stage's value.
+   * @param <D>      the type of the fourth stage's value.
+   * @param <E>      the type of the fifth stage's value.
+   * @param <F>      the type of the sixth stage's value.
+   * @return a stage that completes into the return value of the supplied function.
+   * @since 0.1.0
```
Comment (mattnworb) [inline src/main/java/com/spotify/futures/CompletableFutures.java]: wrong version number?

**Comment by mattnworb** [comment]
I cut a release for `0.3.2` with this in it.

**Pull request merged**